Kernel/Process.cpp

```diff
@@ -1540,6 +1540,11 @@ void Process::crash(int signal, u32 eip)
     } else {
         dbg() << "\033[31;1m" << String::format("%p", eip) << "  (?)\033[0m\n";
     }
+
+    dbg() << "Creating dump of physical memory...";
+    create_core_dump();
```

Comment:
@awesomekling Would you prefer it if I made this some kind of switch from a configuration file (or a Kernel command line switch???)

Reply:
I believe it should be fine to dump setuid programs as long as we ensure the core file is owned by root (or whoever owns the process). Now, of course there's the issue that setuid is not only for root — in effect, a single process runs as multiple users, and it's not always that one of them is strictly more privileged than the other.

Reply:
So this could cause some kind of vulnerable or undefined state?

Reply:
For example, `su` may have user's password in memory when it crashes and we wouldn't want `anon` to be able to inspect the core dump and get the password from there.

Reply:
Ahhh, yep, I thought as much.